intro-HPC/ch_running_batch_jobs.tex

```diff
@@ -40,6 +40,494 @@ \chapter{Running batch jobs}
+    the versioning of software packages. For instance, it does not allow multiple
+    versions to be loaded at same time.
+\item  Takes care of \emph{dependencies}: Another issue arises when one considers
+    library versions and the dependencies they require. Some software requires an
+    older version of a particular library to run correctly (or at all). Hence a
+    variety of version numbers is available for important libraries. Modules
+    typically load the required dependencies automatically.
+\end{enumerate}
+
+This is all managed with the \lstinline|module| command, which is explained in the next sections.
+
+\ifusinglmod
+There is also a shorter \lstinline|ml| command that does exactly the same as the \lstinline|module|
+command and is easier to type. Whenever you see a \lstinline|module| command, you can replace
+\lstinline|module| with \lstinline|ml|.s
+\fi
+
+
+\subsection{Available modules}
+
+A large number of software packages are installed on the \hpc clusters. A
+list of all currently available software can be obtained by typing:
+
+\begin{prompt}
+%\shellcmd{module available}%
+\end{prompt}
+
+It's also possible to execute \lstinline|module av| or \lstinline|module avail|,
+these are shorter to type and will do the same thing.
+
+This will give some output such as:
+
+\inputsite{available-modules}
+
+This gives a full list of software packages that can be loaded.
+
+\strong{The casing of module names is important}: lowercase and uppercase letters
+matter in module names.
+
+\subsection{Organisation of modules in toolchains}
+
+The amount of modules on the VSC systems can be overwhelming, and it is not
+always immediately clear which modules can be loaded safely together if you
+need to combine multiple programs in a single job to get your work done.
+
+Therefore the VSC has defined so-called \strong{toolchains}.
+A toolchain contains a C/C++ and Fortran compiler, a MPI library and some basic
+math libraries for (dense matrix) linear algebra and FFT. Two toolchains are
+defined on most VSC systems. One, the \lstinline|intel| toolchain, consists of the Intel
+compilers, MPI library and math libraries. The other one, the \lstinline|foss| toolchain,
+consists of Open Source components: the GNU compilers, OpenMPI, OpenBLAS and
+the standard LAPACK and ScaLAPACK libraries for the linear algebra operations
+and the FFTW library for FFT. The toolchains are refreshed twice a year,
+which is reflected in their name.
+
+E.g., \texttt{foss/\the\year{}a} is the first version of the \lstinline|foss| toolchain in \the\year.
+
+The toolchains are then used to compile a lot of the software installed on
+the VSC clusters. You can recognise those packages easily as they all contain the
+name of the toolchain after the version number in their name (e.g. \lstinline|Python/2.7.12-intel-2016b|).
+Packages compiled with the same toolchain and toolchain version will typically work
+together well without conflicts.
```

Comment:
That's actually not true at all, since there often are conflicts on the dependencies as well, e.g. Python 2 vs 3. We pay more attention to this recently in Ghent, but only for the most recent toolchains (`2018a` & newer)

Please rephrase: `Only packages compiled with the same toolchain name and version can work together without conflicts.`